docs/git/terminal/lesson.md

```diff
@@ -0,0 +1,216 @@
+
+When the Spotlight Search bar pops up on your screen, type `terminal.app` and hit `Return`. Or you can click the Terminal.app icon that appears.
+
+Terminal will launch, and you’ll be ready to go.
+
+#### Ubuntu
+
+You’ll find the Terminal application included with your other installed applications. To find them, click the `Show Applications` button at the bottom-left corner of your screen, on the dash bar.
+
+Type `Terminal` and press Enter to find and launch the Terminal shortcut. You can also locate the Terminal icon in the list of all applications that appears here and click it.
+
+### Exercise (5 minutes)
+
+:::tip Exercise
+Using the above instructions, open the terminal now.
+
+Having issues? Can't open the terminal? Ask for help on your Slack channel **now**!.
+:::
+
+## 2) What is a GUI?
+
+GUI stands for **G**raphical **U**ser **I**nterface.
+
+A GUI is a useful way to make computers easier to use for general users of software. An example of a GUI is the Windows Desktop.
+
+![Windows 7 Desktop - Annotated](assets/windows-desktop.jpg)
+
+In a GUI, there a visual elements (buttons, text, windows etc) that a user can interact with to complete tasks.
+
+### Self-Study Exercise (15 minutes)
+
+:::tip Exercise
+Research the following questions
+
+- What is a GUI useful?
+- What are three examples of a GUI?
+- When were GUIs invented?
+- What came before GUIs?
+
+Anything you don't understand? Ask your questions in your Slack channel.
+:::
+
+## 3) What is a CLI (or Terminal)?
+
+CLI stands for **C**ommand **L**ine **I**nterface. An example of a CLI is Git Bash.
+
+![Example of a Terminal](assets/terminal-example.png)
+
+The terminal exists to take commands that you write and turn them into something that the computer can **do**.
+
+Many of the tasks that you complete using a GUI can also be done using the command line. For example,
+
+- Creating a file
+- Creating a folder
+- Looking at the contents of a file
+- Connecting to the internet
+- Opening programs
+
+You can write `commands` to your computer by wring text after the `$` sign in your terminal window and pressing `enter`.
+
+![An Empty Terminal Window](assets/terminal-empty.png)
+
+However, you can't just type anything! Your computer only understand commands in a specific format
+
+![Terminal with an error](assets/terminal-error.png)
+
+If you type something that the computer doesn't understand, you'll see the above error.
+
+### Self-Study Exercise (5 minutes)
+
+<!--
+This is built using the Tabs component that is included in docusaurus
+https://v2.docusaurus.io/docs/markdown-features/#multi-language-support-code-blocks
+-->
+
+<Tabs
+defaultValue="exercise"
+values={[{ label: "Exercise", value: "exercise", },{ label:"Answer", value: "answer", },]}>
+<TabItem value="exercise">
+
+:::tip Exercise
+Make the command line do something!
+
+It's really really hard to break something in the terminal by mistake.
```

Comment:
I'm slightly wary of saying this... Something like `sudo rm -rf /` is really easy to do, and if they're maybe going to Google random stuff to run in their terminal, it's not impossible they'll run into something like this...

Reply:
Fair enough! I'll replace it with a list of words